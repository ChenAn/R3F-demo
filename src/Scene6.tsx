/** @jsxImportSource @emotion/react */
import React from "react";
import * as Fiber from '@react-three/fiber';
import { css } from '@emotion/react';
import MainScene6 from "./components/MainScene6";

const styles = {
  container: css({
    height: "90vmin",
    width: "100%",
    flexDirection: "column",
    alignItems: "center",
  }),
  canvas: css({
    width: "100%",
    height: "100%",
    touchAction: "auto",
  }),
};

const Scene6 = () => {
  return (
    <div css={styles.container}>
      <p>テスト用にテキスト</p>
      <Fiber.Canvas css={styles.canvas} onCreated={state => state.gl.setClearColor("black")} >
        <MainScene6/>
      </Fiber.Canvas>
    </div>
  );
};

export default Scene6;
