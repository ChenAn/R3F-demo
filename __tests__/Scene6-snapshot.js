import React from 'react';
import { render } from '@testing-library/react';
import Scene6 from '@src/Scene6'

describe('Scene6', () => {
  let originResizeObserver;
  beforeEach(() => {
    originResizeObserver = global.ResizeObserver;
    global.ResizeObserver = jest.fn().mockImplementation(() => ({
      observe: jest.fn(),
      unobserve: jest.fn(),
      disconnect: jest.fn(),
    }));
  });

  afterEach(() => {
    global.ResizeObserver = originResizeObserver;
  });

  it('初期表示', () => {
    const { asFragment } = render(
      <Scene6/>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
