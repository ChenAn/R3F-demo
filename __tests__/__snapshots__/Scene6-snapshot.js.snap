// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scene6 初期表示 1`] = `
<DocumentFragment>
  <div
    class="css-1np4oav-Scene6"
  >
    <p>
      テスト用にテキスト
    </p>
    <div
      class="css-1atx4qx-Scene6"
      style="position: relative; width: 100%; height: 100%; overflow: hidden;"
    >
      <canvas
        style="display: block;"
      />
    </div>
  </div>
</DocumentFragment>
`;
